### PR TITLE
Base connections pool

### PR DESCRIPTION
**Pull request by diegoreis42**
This pull request address the issue https://github.com/basementdevs/scylla-studio/issues/6 by:

1.  Implementing a modal component to create a new connection;
2. Adding a form that captures basic connection details, with validation handled by `zod` and `react-hook-form`.
3. Using SQLite to persist connection information and populate the connections table.

The design supports both Light and Dark themes, maintaining consistency with the overall component styles. Feedback and suggestions for improvement are welcome.

I have some issues using the `shadcn` Dialog component so I created a custom one. A good future improvement could be using the shadcs's component to improve consistency across the codebase.

### CODE REVIEW

File: src/app/(main)/connections/_components/modal.tsx [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+
+  const handleSave = () => {
+    const validationErrors = validateForm();
+
+    if (Object.keys(validationErrors).length > 0) {
+    } else {
+      startTransition(async () => {
+      });
+    }
+  };
```
Comment (Daniel-Boll) [inline src/app/(main)/connections/_components/modal.tsx:62]: Esse código parece estranho, qual o propósito?

**Comment by Daniel-Boll** [comment]
Apart from that looking fine

File: src/app/(main)/connections/_components/modal.tsx (hunk 1 above)
Comment (diegoreis42) [reply src/app/(main)/connections/_components/modal.tsx:62]: Foi descuido meu, commitei antes de finalizar a parte de erros e ficou faltando esse tratamento. Ele basicamente é para tratar as validações do forms.

**Review by Daniel-Boll — APPROVED** [review]

File: src/app/(main)/connections/_components/modal.tsx (hunk 1 above)
Comment (Daniel-Boll) [reply src/app/(main)/connections/_components/modal.tsx]: Entendi, beleza então. Acho que num futuro dá pra integrar com um `react-hook-form` e `zod` ou `yup` pra essas validações.

**Review by DanielHe4rt — APPROVED** [review]
The main feature covers what is expected but here's some input:

- There's no description at the Pull Request itself besides the title, please update it with a brief explanation on what you did/implemented.
- Since the project is mainly using Shadcn, why not use their [Form Component](https://ui.shadcn.com/docs/components/form)?
- The Tailwind implementation should cover at least one theme.

In case you're not familiarized with Tailwind, here's some tips:

## Theming

To style different themes, you should identify which theme exists and bring the prefix together with a class of your choice, e.g: 

```html
<div className="bg-white dark:bg-black"></div>
```

While white theme is active (no prefix), it will ignore the `dark:` rule. You can check different components or inspect directly some components in the source to have a better perspective on how this theming is currently made by Shadcn.

## Validation

Using the zod implementation you can easily validate your form, here's a snippet on it:

```js
const ip = z.string().ip();

ip.parse("192.168.1.1"); // pass
ip.parse("84d5:51a0:9114:1855:4cfa:f2d7:1f12:7003"); // pass
ip.parse("84d5:51a0:9114:1855:4cfa:f2d7:1f12:192.168.1.1"); // pass

ip.parse("256.1.1.1"); // fail
ip.parse("84d5:51a0:9114:gggg:4cfa:f2d7:1f12:7003"); // fail
```

> I highly encourage you to learn more about Tailwind and Zod to enhance this Pull Request.

Let me know if you're missing any information or if you want to do some pair coding. I'm not the best JS/TS developer however I can try to help somehow.

File: src/app/(main)/connections/_components/modal.tsx [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+import { ReactNode, useState, useTransition } from "react";
+import { FormProvider, useForm } from "react-hook-form";
+import { z } from "zod";
+
+const formSchema = z.object({
+  name: z.string().min(1, { message: "Name is required." }),
+  host: z
+    .string()
+    .min(1, { message: "Host is required." })
+    .regex(/^[a-zA-Z0-9.-]+$/, "Invalid host format."),
```
Comment (DanielHe4rt) [inline src/app/(main)/connections/_components/modal.tsx:25]:
```suggestion
  host: z
    .string()
    .ip()
    .min(1, { message: "Host is required." })
    .regex(/^[a-zA-Z0-9.-]+$/, "Invalid host format."),
```

**Review by DanielHe4rt — APPROVED** [review]
That's way better in all aspects! Thanks for your contribution.

cc: @Matozinho @Daniel-Boll

File: src/app/(main)/connections/_components/modal.tsx (hunk 2 above)
Comment (Daniel-Boll) [reply src/app/(main)/connections/_components/modal.tsx:25]: Se for usar o `z.string().ip()` aí não precisa do `regex` e talvez nem o `min`

**Review by Daniel-Boll — APPROVED** [review]

**Pull request merged**